### PR TITLE
Add Take and Drop function

### PR DESCRIPTION
**Pull request by bloderxd**
### What's this?

This Pull Request serves just to add two more functions to Seq class.
#### Take

Take is a function that has goal to get N elements in a sequence. Example:

```
public ArrayList takeExample() {
    Integer[] numbers = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
    return new Seq<Integer>(numbers).take(3).toArrayList();
} 

=> [1, 2, 3]
```
#### Drop

Drop is a function that has goal to remove N elements in a sequence. Example:

```
public ArrayList dropExample() {
    Integer[] numbers = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
    return new Seq<Integer>(numbers).drop(7).toArrayList();
}

=> [8, 9, 10]
```
### Questions

Hey @ppamorim @vitorprado @haskellcamargo I saw your tail function ( you did it with subList ) and I had tried to do these functions with subList, but I couldn't call them with toArrayList Seq function, because a subList can't be casted to arrayList, then I want to know how do you call this tail function?
### @vitorprado

Ok prado, you know I'm beginner in tests, I don't know if this is the best way to do these tests then, be friendly with me :shipit: ( <--- I love this emoji hehe )

### CODE REVIEW

**Comment by bloderxd** [comment]
Sorry about travis CI I'll fix it ( forgot to configure android studio ).

**Comment by ppamorim** [comment]
I like it.

File: seq/src/test/java/com/jmonad/seq/SeqDropTest.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.junit.Test;
+
+public class SeqDropTest {
+
+  @Test public void dropListElementsTest() {
+    Integer[] numbers = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+
+    Seq<Integer> elements = new Seq<Integer>(numbers);
+
+    assert elements.drop(5).toArrayList().toString().equals("[6, 7, 8, 9, 10]");
```
Comment (vitorprado) [inline seq/src/test/java/com/jmonad/seq/SeqDropTest.java:11]: I think you can split this assertions into other tests. Make just one assertion in each test, and use a self explanatory name.. saying what you're testing, the input and the expected.. Sometimes you're test expect an error, and you should explain it in the test. ok?

Comment (bloderxd) [inline seq/src/test/java/com/jmonad/seq/SeqDropTest.java:11]: Ok, I'll do it.

**Comment by vitorprado** [comment]
@bloder You did a good job little boy!! I just ask you to split your tests.

**Comment by bloderxd** [comment]
Ok, just a question about tail function, I don't know how you call it but, it's not working with toArrayList because you did with subList and can't be casted to an ArrayList, I can fix it, do you want or you call tail function in a different way that I didn't see?

**Comment by vitorprado** [comment]
@bloder how about finish this PR and after that take a look into `tail` function? maybe we can fix it and write some tests, but in another PR.

**Comment by bloderxd** [comment]
Ok, I'll handle it.

**Comment by bloderxd** [comment]
@vitorprado Ok, did it.

**Comment by vitorprado** [comment]
👍 👍 :shipit:

File: seq/src/main/java/com/jmonad/seq/Seq.java [hunk 2]
```diff
@@ -231,6 +231,34 @@ public boolean empty() {
   }
 
   /**
+   * Returns the first n elements of the sequence
+   * @param amount
+   * @return Seq<T>
+   */
+  public Seq<T> take(int amount) {
+    Seq<T> buffer = new Seq<T>();
```
Comment (racerxdl) [inline seq/src/main/java/com/jmonad/seq/Seq.java]: You can omit the T on the new part. Actually the IDE Lint warns about it.

File: seq/src/main/java/com/jmonad/seq/Seq.java [hunk 3]
```diff
@@ -231,6 +231,34 @@ public boolean empty() {
...
+    if (amount <= 0) return buffer;
+    if (amount > this.list.size()) return new Seq<T>(this.list);
```
Comment (racerxdl) [inline seq/src/main/java/com/jmonad/seq/Seq.java]: Same here.

File: seq/src/main/java/com/jmonad/seq/Seq.java [hunk 4]
```diff
@@ -231,6 +231,34 @@ public boolean empty() {
...
+    return buffer;
+  }
+
+  /**
+   * Remove the first n elements of the sequence
+   * @param amount
+   * @return Seq<T>
+   */
+  public Seq<T> drop(int amount) {
+    Seq<T> buffer = new Seq<T>();
```
Comment (racerxdl) [inline seq/src/main/java/com/jmonad/seq/Seq.java]: Same here

File: seq/src/main/java/com/jmonad/seq/Seq.java [hunk 5]
```diff
@@ -231,6 +231,34 @@ public boolean empty() {
...
+    if (amount <= 0) return new Seq<T>(this.list);
```
Comment (racerxdl) [inline seq/src/main/java/com/jmonad/seq/Seq.java]: Same here

**Comment by racerxdl** [comment]
Looks ok to me. I just made some comments about initializing Seq. You don't need to specify the T type in the right side of the initialization.

```
Seq<T> mySeq = new Seq<>();
```

**Comment by bloderxd** [comment]
@racerxdl Thank you for your comment, I know that, I just did it because other seq functions are doing in this way.

**Comment by racerxdl** [comment]
Makes sense @bloder. btw I changed this on my PR #1

**Comment by bloderxd** [comment]
@racerxdl Sorry, I didn't see your PR, I'll fix it.

**Comment by racerxdl** [comment]
For me it can be merged. Lets see if @ppamorim @vitorprado @haskellcamargo agrees with that. I think this functions are usefull.

**Pull request merged**